src/main/java/org/apache/maven/buildcache/CacheControllerImpl.java

```diff
@@ -503,6 +503,7 @@ public void save(
                     }
                 }
             } else {
+                localCache.beforeSave(context);
```

Comment:
as this will done for both if and else maybe better to write this only once before the if?

Reply:
Oh sorry, I should have left a comment in the MR intially ...

While checking existing issues, I stumbled upon MBUILDCACHE-71 (only publish buildinfo after optional artifacts were pushed) and tried to pull both calls (::beforeSave and ::saveBuildInfo) actually down/below the "if", but that makes Issue67Test actually fail.

I'll try to have a look what the expectations are on Issue67Test, but my time is a bit limited. So is the proposoal to move both statements above the "if" and dropping the "else branch" acceptable for the moment?